### ISSUE TITLE
Danger list suggestion concerning Ratings given

### ISSUE THREAD

**Issue by pfefferminzz** [ludumdare/ludumdare#971]
Games with Ratings Given: 0 should never be able to be on the first Danger page if there are games with Ratings Given > 0. A simple approach would be:

Sort the list as you like (Classic, Smart, Karma), but at the end it is divided in two chunks:
All games with Ratings Given > Ratings Recieved are shown before all games with RG <= RR at all times.

The Reason behind this? A game deserves as many ratings as it has given to others.

An example with three chunks:
TOP: RG > RR
MIDDLE: RG * 2 > RR (equals 10 RG 20 RR at the end)
BOTTOM: RG * 2 <= RG

**Comment by Cerno-b** [comment]
This is related to my post: https://github.com/ludumdare/ludumdare/issues/967
I suggested multiplying the number of ratings with the game's Coolness, which is much easier.

That way a game with 17 ratings and a coolness of 20 will appear higher on the list than a game with 19 ratings and a coolness of -20. Of course, games with more than 20 ratings do not appear on the list.

In general I totally agree with the basic idea though, as I think it doesn't matter if you are one point short of reaching 20 if you haven't rated: If there is someone who did rate and has 15 votes missing, they should get preference.

**Comment by pfefferminzz** [comment]
Originally i posted this under your issue, but i created a new issue, because my suggested execution differs a bit. But essentially, we want the same thing, of course. :)

**Comment by pfefferminzz** [comment]
![ratingsgiven](https://cloud.githubusercontent.com/assets/5459440/26274825/70a07b86-3d53-11e7-8162-efe78d3fe9c8.png)

Crosspost from the news discussion here: https://ldjam.com/events/ludum-dare/38/ludum-dare-dot-com/short-on-ratings-friday-new-features

**Comment by local-minimum** [comment]
I also discussed this in on http://gitter.im/ludumdare/ludumdare

Suggestion for simple rescue metric that will benefit those that need rescue and also are helping out themselves:

```php
if ($votes_on_my_game >= 20) {
    $rescue_score = -1;
} else {
    $rescue_score = $votes_on_my_game + $c * $votes_by_team_on_other_games; 
}

sort descending;
```

I would suggest having `$c` be around 3-4.

Benefit of being really simple math and easy to understand what happens.

Effects:

* Will focus on easy to rescue
* Will quickly promote those that need rescue and contribute to the top of rescue page.
* Hopefully stop desperate people spamming since they'll quickly see themselves on top. Should also hopefully make desperate give fair and respectful ratings on other's games
* Potential negative effect is that it won't keep greedy needy people contributing very long because they will quickly be rescued after having votes on some games themselves.

Though slightly off topic I also think something following:

I found where you toggle default filter: `let DefaultFilter = 'smart';//'danger';//'smart';` and I'm thinking one way of doing it would be `let DefaultFilter = Math.random() < lerp(1.0, 0.4, VotingPeriodProgress) ? 'smart' : 'danger';`

**Comment by zwrawr** [comment]
I like @local-minimum suggestion but i'd go with a smaller `$c` between `1/2 <= $c <= 2`

**Comment by local-minimum** [comment]
@zwrawr yeah I guess that would do too, I basically just wanted some quick feedback to the desperate user that they are moving up in the `danger` list. Oh, it would be cool if there was a nice way of polling where does my game rank in the different lists and each time any rank shifts give a flash of:

> **Danger:** ⬆️ 10 | **Smart:** ⬇️ 3

**Comment by zwrawr** [comment]
Now thats cool idea. 
But we don't really know where a game actually is in the feed most of the time, because of the way our batching works.

**Comment by local-minimum** [comment]
But, we do know each time the stats have been reworked right? So if client side keeps track of that and asks if new batch of stats is out and then gets the data?

**Comment by Cerno-b** [comment]
@local-minimum I like that, it's similar to what I proposed but much clearer. My version had rating*coolness, but those values would have to be normalized which is overly complicated.

**Comment by zwrawr** [comment]
But the client cannot get all the games, to figure out where it is.
It's totally possible server side im just saying that at the moment we don't keep track of where nodes are in the feeds.

**Comment by local-minimum** [comment]
Ah, I misunderstood you but you could, though it would require some api calls use:

https://api.ldjam.com/vx/node/feed/1/danger/item/game/compo+jam+craft+release?offset=7&limit=17

if you somehow know your current offset, aka placement, you can search on either side of that for your id.
of course it would be nicer, and less expensive on server side if the server could know current rank or at least allow for:

https://api.ldjam.com/vx/node/feed/1/danger/item/whatsmy/[gamenode]/compo+jam+craft+release

or however the api should be.
